privacyidea/static/components/token/controllers/tokenDetailController.js

```diff
@@ -190,9 +190,42 @@ myApp.controller("tokenDetailController", function ($scope,
         }, $scope.get);
     };
 
-    $scope.delete = function () {
-        TokenFactory.delete($scope.tokenSerial, $scope.return_to);
+    $scope.deleteTokenAsk = function() {
+            $('#dialogTokenDelete').modal();
+        var tokenType = $scope.token.info.tokenkind;
+        if (tokenType == "hardware"){
+        $('#dialogTokenDelete').modal();
+             } else {
+             TokenFactory.delete($scope.tokenSerial, $scope.return_to,
```

Comment:
The third parameter here does not do anything (see explanation below). What are you trying to do here? We already have a function `$scope.delete()` that calls `TokenFactory.delete()` with the correct parameters. Unless there is a specific reason why you need the extra code, please just call `$scope.delete()`.